ext/web/16_report_error.js

```diff
@@ -0,0 +1,77 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+"use strict";
+
+((window) => {
+  const core = window.Deno.core;
+  const { ErrorEvent } = window;
+  const { Error, TypeError } = window.__bootstrap.primordials;
+  const webidl = window.__bootstrap.webidl;
+
+  let errorReported = false;
+
+  function handleReportErrorMacrotask() {
+    if (errorReported) {
+      errorReported = false;
+      throw new Error("Unhandled error event from 'reportError()'.");
+    }
+    return true;
+  }
+
+  let printException = undefined;
+
+  function setPrintException(fn) {
+    printException = fn;
+  }
+
+  function checkThis(thisArg) {
+    if (thisArg !== null && thisArg !== undefined && thisArg !== globalThis) {
+      throw new TypeError("Illegal invocation");
+    }
+  }
+
+  function reportError(error) {
+    checkThis(this);
```

Comment:
We dont do this for `fetch`, so not sure how correct this is. @lucacasonato

Reply:
Yeah me neither, I took it from `setTimeout()`